tensorboard/webapp/runs/views/runs_table/runs_group_menu_button_component.ts

```diff
@@ -36,27 +36,37 @@ export class RunsGroupMenuButtonComponent {
   @Input() experimentIds!: string[];
   @Input() regexString!: string;
   @Input() selectedGroupBy!: GroupBy;
+  @Input() enableColorByExperiment!: boolean;
 
   @Output()
   onGroupByChange = new EventEmitter<GroupBy>();
 
   constructor(private readonly dialog: MatDialog) {}
 
   onRegexStringEdit() {
-    // data pass in the experiment id
-    const dialogRef = this.dialog.open(RegexEditDialogContainer, {
+    this.dialog.open(RegexEditDialogContainer, {
       maxHeight: '95vh',
       maxWidth: '80vw',
-      data: {experimentIds: this.experimentIds},
+      data: {
+        experimentIds: this.experimentIds,
+      },
     });
   }
 
   onGroupByRegexClick() {
     if (!this.regexString) {
       this.onRegexStringEdit();
     } else {
+      let next_key = GroupByKey.REGEX;
+      // If feature is enabled user can toggle between run and experiment name regex matching.
+      if (this.enableColorByExperiment) {
```

Comment:
nit ) This eliminates the need for Trililogic.
```ts
if( this.enableColorByExperiment && this.selectedGroupBy.key === GroupByKey.REGEX) {
  next_key = GroupByKey.REGEX_BY_EXP
}
```

Reply:
That's neater. Thanks!